homeassistant/components/octoprint/config_flow.py

```diff
@@ -0,0 +1,197 @@
+"""Config flow for OctoPrint integration."""
+import logging
+
+from pyoctoprintapi import ApiError, OctoprintClient
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PATH,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "OctoPrint"
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=80): cv.port,
+        vol.Optional(CONF_PATH, default="/"): cv.string,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+STEP_IMPORT_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=80): cv.port,
+        vol.Optional(CONF_PATH, default="/"): cv.string,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def _schema_with_defaults(host=None, port=80, path="/"):
+    return vol.Schema(
+        {
+            vol.Required(CONF_USERNAME): cv.string,
+            vol.Required(CONF_HOST, default=host): cv.string,
+            vol.Optional(CONF_PORT, default=port): cv.port,
+            vol.Optional(CONF_PATH, default=path): cv.string,
+            vol.Optional(CONF_SSL, default=False): cv.boolean,
+            vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
```

Comment:
Do we need to allow configuring name of the server? Normally we try to avoid that kind of option in config flows. The user will be able to change name of the device and entities anyway.

Reply:
This was existing, I can drop support for it however

Reply:
What should I be having the sensor names be prefixed with in these cases? I don't believe there is a name supplied from octoprint. Should they just be name "sensor.printer_status"? I think this would cause name collisions if adding multiple printers?

Reply:
I would just prefix it with "OctoPrint Printer". Entity IDs don't collide, Home Assistant will take care of that, and with the unique ID, a user can just rename both the name and ID of the entity ID to their liking.

Reply:
Done